prow/tide/status.go

```diff
@@ -380,70 +372,34 @@ func (sc *statusController) search() []PullRequest {
 	// adding statuses to excluded repos.
 	orgExceptions, repos := sc.config().Tide.Queries.OrgExceptionsAndRepos()
 	orgs := sets.StringKeySet(orgExceptions)
-	freshOrgs, freshRepos := orgs.Difference(sc.trackedOrgs), repos.Difference(sc.trackedRepos)
-	oldOrgs, oldRepos := sc.trackedOrgs.Difference(orgs), sc.trackedRepos.Difference(repos)
-	// Stop tracking orgs and repos that aren't queried this loop.
-	sc.trackedOrgs.Delete(oldOrgs.UnsortedList()...)
-	sc.trackedRepos.Delete(oldRepos.UnsortedList()...)
-	// Determine the query for tracked PRs now before we modify 'trackedOrgs' and 'trackedRepos'.
-	var trackedQuery string
-	if sc.trackedOrgs.Len() > 0 || sc.trackedRepos.Len() > 0 {
-		trackedQuery = openPRsQuery(sc.trackedOrgs.UnsortedList(), sc.trackedRepos.UnsortedList(), orgExceptions)
+	query := openPRsQuery(orgs.List(), repos.List(), orgExceptions)
+	if query != sc.previousQuery {
+		// Query changed and/or tide restarted, recompute everything
+		sc.latestPR = time.Time{}
+		sc.previousQuery = query
 	}
-	queryStartTime := time.Now()
-
-	var lock sync.Mutex
-	var wg sync.WaitGroup
-	var allPRs []PullRequest
-	// Query fresh orgs and repos individually and since the beginning of time.
-	// These queries are larger and more likely to fail so we query for targets individually.
-	singleTargetSearch := func(query, target string, tracked sets.String) {
-		defer wg.Done()
-		searcher := newSearchExecutor(context.Background(), sc.ghc, sc.logger, query)
-		prs, err := searcher.search()
-		if err != nil {
-			sc.logger.WithError(err).Errorf("Searching for open PRs in %s.", target)
-			return
+	start := time.Now()
+
+	prs, err := search(sc.ghc.Query, sc.logger, query, sc.latestPR, start)
```

Comment:
We should probably have at least a small amount of time window overlap. We were previously overlapping by 30 seconds by subtracting 30 seconds from the query window start time.
```
	// We offset for 30 seconds of overlap because GitHub sometimes doesn't
	// include recently changed/new PRs in the query results.
```

Reply:
Done